lib/gem_tracker/ci/github.rb

```diff
@@ -64,15 +64,18 @@ def get_repository
 
   # https://docs.github.com/en/rest/reference/actions#list-jobs-for-a-workflow-run
   def get_run_jobs(jobs_url)
-    jobs_url = "#{jobs_url}?per_page=100"
-    request(jobs_url) do |response|
-      json = JSON.parse(response.body)
-      jobs = json.fetch("jobs") { pp json; raise "Couldn't find runs jobs" }
-      if jobs.size != json['total_count']
-        raise "request did not return all jobs: #{jobs.size} of #{json['total_count']}"
+    urls_enum = (1..).lazy.map { |page| "#{jobs_url}?per_page=100&page=#{page}" }
+    job_pages_enum = urls_enum.map do |url|
+      request(url) do |response|
+        json = JSON.parse(response.body)
+        jobs = json.fetch("jobs") { pp json; raise "Couldn't find runs jobs" }
+        jobs
       end
-      jobs
     end
+
+    job_pages = job_pages_enum.take_while { |job_page| !job_page.empty? }.to_a
```

Comment:
This likely makes an extra request, maybe we can find the total number of jobs somehow to avoid the extra empty page request? Can you check?

Reply:
From the previous code `json['total_count']` has that information.

Reply:
Yep, makes sense. Replaced `!job_page.empty?` with `job_page.size == page_size` and reverted the check of the total size to ensure everything works as expected.